scio-avro/src/main/scala/com/spotify/scio/avro/AvroIO.scala

```diff
@@ -158,8 +169,17 @@ final case class SpecificRecordIO[T <: SpecificRecord: ClassTag: Coder](path: St
    */
   override protected def read(sc: ScioContext, params: ReadP): SCollection[T] = {
     val cls = ScioUtil.classOf[T]
-    val t = beam.AvroIO.read(cls).from(path)
-    sc.applyTransform(t)
+
+    // Adapt Beam to accept datum writer + coders
+    val source: AvroSource[T] = AvroSource
+      .from(StaticValueProvider.of(path))
+      .withEmptyMatchTreatment(EmptyMatchTreatment.DISALLOW)
+      .withDatumReaderFactory(new SpecificDatumReaderFactory[T]())
+      .withSchema(cls)
+      .withParseFn(_.asInstanceOf[T], AvroCoder.of(cls, false))
```

Comment:
Ideally, beam [AvroIO.Read](https://github.com/apache/beam/blob/b78de493a4a5f56cf8339ab2f174cc25cc57beec/sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java#L591) should be updated so it can accept a `DatumReaderFactory` and a `Coder`